style_guides/pl_sql_tables.md

```diff
@@ -0,0 +1,64 @@
+# Database Tables
+
+## Table Names
+
+Tables should be given a meaningful name that describes the purpose of the table. All table names should be plural and where the table name contains more than one word only the last should be plural. 
+
+Table names should be underscore separated. 
+
+In a shared database schema ( e.g. BANDEV ) , tables that relate to a specific domain should use a prefix to relate them to one another. This is not required in a schema explicitly created for a specific domain ( e.g. TRAVEL_COVER )
+
+Some example table names in a shared database schema: 
+
+```
+timetable_programmes
+timetable_programme_years
+timetable_programme_year_modules
+timetable_export_log
+```
+
+### General Guidance
+
+**Don't prefix table names with programme or project names.** For  example, StARS or New Core. Always name tables according to domain ( in a shared schema ) and purpose. 
+
+**When creating new tables ensure that you get the right balance of brevity and meaningfulness**. Don't create tables with unnecessarily long names or names that are so short that it is difficult to understand their purpose. 
+
+
+## Constraints / Indexes
+
+Table constraints should be named as followed. Where an index is manually defined it should follow the convention below. 
+
+```
+pk_<table>                 -- Primary Key
+fk_<table>_<optional_text> -- Foreign Key
+idx_<table>_<description>  -- Manually defined index
+```
+
+## Table comments
+
+Table and column comments should be included in the creation script for all new tables. 
+
+```sql
+COMMENT ON TABLE schema.table IS 'Overall table description';
+
+
+COMMENT ON COLUMN schema.table.column1 IS 'Column 1 description';
+COMMENT ON COLUMN schema.table.column2 IS 'Column 2 description';

```

Comment:
Should we define schema here? do we fall into the issue that deploying into another schema then does not work like test schemas for deployment script dry runs

Reply:
Do we have test schemas where we have deployed into ? I like to prefix with schema personally as it prevents you accidently deploying to the wrong schema ( I have done this ) but also it also indicates you know where this should be deployed. So far to date anything I have worked on/deployed has never been deployed to more than one schema. But if we are planning on deploying things into multiple like you say then maybe what you say makes sense.